jena-arq/src/main/java/org/apache/jena/sparql/algebra/optimize/TransformPropertyFunction.java

```diff
@@ -27,93 +27,75 @@
 import org.apache.jena.sparql.util.Context ;
 
 /** Rewrite to replace a property function property with the call to the property function implementation */
-public class TransformPropertyFunction extends TransformCopy
-{
-    private final Context context ;
-    private final boolean doingMagicProperties ;
-    private final PropertyFunctionRegistry registry ;
-    
-    /** Apply the property function transformation. 
-     *  <p>
-     *  The {@code context} provide the place to find the property function registry.
-     *  A custom one canbe supplied using 
-     *  {@link ARQConstants#registryPropertyFunctions}
-     *  <p>
-     *  See {@link PropertyFunctionRegistry#chooseRegistry} for the full decision process.
-     *  <p>
-     *  In addition, {@link ARQ#enablePropertyFunctions} must be true (this is the default). 
+public class TransformPropertyFunction extends TransformCopy {
+    private final Context context;
+    private final boolean doingPropertyFunctions;
+    private final PropertyFunctionRegistry registry;
+
+    /**
+     * Apply the property function transformation.
+     * <p>
+     * The {@code context} provides the place to find the property function registry.
+     * A custom one can be supplied using
+     * {@link ARQConstants#registryPropertyFunctions}
+     * <p>
+     * See {@link PropertyFunctionRegistry#chooseRegistry} for the full decision
+     * process.
+     * <p>
+     * If {@link ARQ#enablePropertyFunctions} is false, then property functions are
+     * not enabled and remain as plain triples. For example, this is set false by "strict mode"
      */
     public static Op transform(Op op, Context context) {
-        Transform t = new TransformPropertyFunction(context) ;
-        return Transformer.transform(t, op) ; 
+        Transform t = new TransformPropertyFunction(context);
+        return Transformer.transform(t, op);
     }
-    
-    public TransformPropertyFunction(Context context)
-    {
-        this.context = context ;  
-        doingMagicProperties = context.isTrue(ARQ.enablePropertyFunctions) ;
-        registry = PropertyFunctionRegistry.chooseRegistry(context) ;
+
+    public TransformPropertyFunction(Context context) {
+        this.context = context;
+        boolean enabled = context.isTrueOrUndef(ARQ.enablePropertyFunctions);
+        registry = PropertyFunctionRegistry.chooseRegistry(context);
+        if ( registry == null )
+            enabled = false;
+        doingPropertyFunctions = enabled;
```

Comment:
```suggestion
        registry = PropertyFunctionRegistry.chooseRegistry(context);
        doingPropertyFunctions = context.isTrueOrUndef(ARQ.enablePropertyFunctions) && registry != null;
```

I think this should have the same behavior? And `doingMagicProperties` was a cool name :tophat: :hatching_chick: :magic_wand: , but +1 on renaming that too.

Reply:
Yes - and added tests to cover all the setup cases.